.github/workflows/push-auto-generate.yml

```diff
@@ -0,0 +1,94 @@
+name: BRM / Automaticly Triggered BRM Generation
+
+on:
+  push:
+    branches-ignore:
+      - main
+    paths:
+      - modules/**.bicep
+
+jobs:
+  check-secret:
+    runs-on: ubuntu-latest
+    outputs:
+      my-key: ${{ steps.my-key.outputs.defined }}
+    steps:
+      - id: my-key
+        if: "${{ env.MY_KEY != '' }}"
+        run: echo "::set-output name=defined::true"
+        env:
+          MY_KEY: ${{ secrets.PAT }}
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: [check-secret]
+    if: needs.check-secret.outputs.my-key == 'true'
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0 # avoid shallow clone so nbgv can do its work.
+          token: ${{ secrets.PAT }} # A PAT must be used to re-trigger workflows after commit. This PAT should NOT have the workflow scope.
+
+      - name: Find Changed Folder
+        id: changed_folders
+        run: |
+          git status
+          # Get path of last file checked in
+          pwd
+          cd $GITHUB_WORKSPACE
+          pwd
+          LAST_PATH=$( git log --pretty="" --name-only -n 100 origin/main... -- | fgrep -v ".github" | head -1 )
+          echo Last file modified: $LAST_PATH
+          # Look for the main template file in this file's path or parents
+          SAMPLEFOLDER_PATH=$( dirname "$LAST_PATH" )
+          echo Last folder modified: $SAMPLEFOLDER_PATH
+          TESTFOLDER_PATH=$SAMPLEFOLDER_PATH
+          FOUNDFOLDER_PATH=
+          while [ "$TESTFOLDER_PATH" != "." ]
+          do
+            echo "Looking for main template in $TESTFOLDER_PATH"
+            MAINBICEP_PATH=$TESTFOLDER_PATH/main.bicep
+            AZDEPLOYJSON_PATH=$TESTFOLDER_PATH/main.json
+            if [ -f "$MAINBICEP_PATH" ] || [ -f "$AZDEPLOYJSON_PATH" ]; then
+              FOUNDFOLDER_PATH=$TESTFOLDER_PATH
+              echo Found main template in $FOUNDFOLDER_PATH
+              break
+            fi
+            TESTFOLDER_PATH=$( dirname "$TESTFOLDER_PATH" )
+          done
+          if [ ! $FOUNDFOLDER_PATH ]; then
+              echo "Could not find main.bicep or azdeploy.json in folder or parents of $SAMPLEFOLDER_PATH" 1>&2
+              exit 1
+          fi
+          echo "SAMPLEFOLDER_PATH=$FOUNDFOLDER_PATH" >> $GITHUB_ENV
+          echo "MAINBICEP_PATH=$MAINBICEP_PATH" >> $GITHUB_ENV
+          echo "AZDEPLOYJSON_PATH=$AZDEPLOYJSON_PATH" >> $GITHUB_ENV
+          echo "SAMPLEFOLDER_PATH=$FOUNDFOLDER_PATH" >> $GITHUB_OUTPUT
+          echo "MAINBICEP_PATH=$MAINBICEP_PATH" >> $GITHUB_OUTPUT
+          echo "AZDEPLOYJSON_PATH=$AZDEPLOYJSON_PATH" >> $GITHUB_OUTPUT
+
+          OLD_TAG=$(cut -d/ -f2-3 <<< $FOUNDFOLDER_PATH)
+          echo "OLD_TAG=$OLD_TAG" >> $GITHUB_OUTPUT
```

Comment:
After reading through the bash script, it seems to me that step can be replaced with https://github.com/Azure/bicep-registry-modules/blob/main/.github/workflows/get-changed-module.yml.